crates/transpiler/src/passes/discretize_rotations.rs

```diff
@@ -0,0 +1,361 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use std::f64::consts::PI;
+
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_circuit::operations::{OperationRef, Param, StandardGate};
+use rustworkx_core::petgraph::stable_graph::NodeIndex;
+
+static ROTATION_GATE_NAMES: [&str; 3] = ["rx", "ry", "rz"];
+
+const PI8: f64 = PI / 8.0;
+const MINIMUM_TOL: f64 = 1e-12;
+
+/// For a given angle, if it is a multiple of PI/4, calculate the multiple mod 16,
+/// Otherwise, return `None`.
+fn is_angle_close_to_multiple_of_pi_4(angle: f64, tol: f64) -> Option<usize> {
+    let closest_ratio = angle * 4.0 / PI;
+    let closest_integer = closest_ratio.round();
+    let closest_angle = closest_integer * PI / 4.0;
+    let theta = angle - closest_angle;
+
+    let tr_over_dim = (theta / 2.).cos();
+    let dim = 2.;
+
+    // fidelity-based tolerance
+    let f_pro = tr_over_dim.abs().powi(2);
+    let gate_fidelity = (dim * f_pro + 1.) / (dim + 1.);
+    if (1. - gate_fidelity).abs() < tol {
+        Some((closest_integer as i64).rem_euclid(16) as usize)
+    } else {
+        None
+    }
+}
+
+/// Gets a rotation gate (RX/RY/RZ) and outputs an equivalent vector of standard gates and
+/// a global phase, when the gate is sufficiently close to Clifford+T/Tdg.
+/// Otherwise, return None.
+/// Note that odd multiples of pi/4 require a single T or Tdg gate
+//  as well as some Clifford gates,
+//  while even multiples of pi/4, or equivalently, integer multiples of pi/2,
+//  can be written using only Clifford gates.
+/// The output contains at most one T or Tdg gate, and an optimal number of
+/// Clifford gates.
+fn try_replace_rotation_by_discrete(
+    gate: StandardGate,
+    angle: f64,
+    tol: f64,
+) -> Option<(Vec<StandardGate>, f64)> {
+    let multiple = is_angle_close_to_multiple_of_pi_4(angle, tol);
```

Comment:
You can return immediately here if `None` is returned here, right? Something like
```rust
let Some(multiple) = is_angle_close_to_multiple_of_pi_4(angle, tol) else {
  return None;
};
```

Reply:
done as part of [5768af2](https://github.com/Qiskit/qiskit/pull/15321/commits/5768af284a6ce3b7fa621121e84dc3629ea18534)

clippy suggests:
`    let multiple = is_angle_close_to_multiple_of_pi_4(angle, tol)?;
`